src/popup/stores/AddTokenStore.ts

```diff
@@ -0,0 +1,109 @@
+import { observable, computed, action, reaction } from 'mobx';
+import { findIndex } from 'lodash';
+
+import AppStore from './AppStore';
+import { MESSAGE_TYPE } from '../../constants';
+import { isValidContractAddressLength } from '../../utils';
+
+const INIT_VALUES = {
+  contractAddress: '',
+  name: '',
+  symbol: '',
+  decimals: undefined,
+  getQRCTokenDetailsFailed: false,
+};
+
+export default class AddTokenStore {
+  @observable public contractAddress?: string;
+  @observable public name?: string;
+  @observable public symbol?: string;
+  @observable public decimals?: number;
+  @observable public getQRCTokenDetailsFailed?: boolean;
+  @computed public get contractAddressFieldError(): string | undefined {
+    return (isValidContractAddressLength(this.contractAddress) && !this.getQRCTokenDetailsFailed)
+    ? undefined : 'Not a valid contract address';
+  }
+  @computed public get buttonDisabled(): boolean {
+    return !this.contractAddress
+    || !this.name
+    || !this.symbol
+    || !this.decimals
+    || !!this.contractAddressFieldError
+    || !!this.tokenAlreadyInListError;
+  }
+  @computed public get tokenAlreadyInListError(): string | undefined {
+
```

Comment:
👾

Reply:
what does this symbol mean?

Reply:
Its a space invader. Meaning delete this extra whitespace lol